src/webview_provider/queryResultPanel.ts

```diff
@@ -352,41 +352,54 @@ export class QueryResultPanel implements WebviewViewProvider {
       this.adapter = adapter;
       this.transmitLoading();
     }
-    try {
-      const output = await queryExecution;
-      await this.transmitDataWrapper(output, query);
-    } catch (exc: any) {
-      if (exc instanceof PythonException) {
-        window.showErrorMessage(
-          "An error occured while trying to execute your query: " +
-            exc.exception.message,
-        );
-        await this.transmitError(
-          {
-            error: {
-              code: -1,
-              message: exc.exception.message,
-              data: JSON.stringify(exc.stack, null, 2),
+    const queryExecution = await queryExecutionPromise;
+    await window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        cancellable: true,
+        title: "Executing sql...",
+      },
+      async (_, token) => {
+        try {
+          token.onCancellationRequested(() => {
+            queryExecution.cancel();
+          });
+          const output = await queryExecution.executeQuery();
+          await this.transmitDataWrapper(output, query);
+        } catch (exc: any) {
+          if (exc instanceof PythonException) {
```

Comment:
can use logging service `dbtTerminal.error` to trigger telemetry event and save to common log file.